[UPDATE] 이슈 상세 페이지 이동, 상세 페이지 버그 수정, 함수 추가, 헤더 추가

pje1740 opened the pull request:
[comment]: <> (PR 태그를 명시해주세요. [ADD] / [UPDATE] / [BUG])

### 요약

- 이슈 상세 페이지 이동
- userPhotoBlock 컴포넌트
- getUserImageLink
- header 추가
- milestone null 체크

### 핵심 로직 및 내용
- 이슈 제목을 클릭하면 이슈 상세 페이지로 넘어갑니다.
- 이슈 상세 페이지로 넘어갈 때, 해당 이슈에 마일스톤이 지정되지 않은 경우 생기던 오류를 수정하였습니다. (null이면 빈 배열로 교체)
- header를 모든 페이지에 추가하기 위해 App.js에서 선언되던 Router를 index.js로 이동하였습니다. (Router로 감싸진 컴포넌트에서는 Link나 history를 사용할 수 없기 때문)
- userPhotoBlock - 유저 사진 출력 컴포넌트
- getUserImageLink - 접속한 유저의 이미지를 cookie에서 추출하여 반환합니다

### 기타 참고 사항

pje1740 committed:
[FEAT] move to issue detail page when issue title is clicked in the issue list page.

[why]
We need to redirect users to the issue detail page when they click the issue title.

[what]
IssueListItem.js - onClickTitle

[etc]

pje1740 committed:
[FEAT] create UserPhotoBlock.js

[why]
This is a component that could be reused in places where user photos need to display.

[what]
UserPhotoBlock.js

[etc]

pje1740 committed:
[FEAT] add getUserImageLink function.

[why]
We need to retrieve user's image link from cookie.

[what]
Source/user.js

[etc]

pje1740 committed:
[FEAT] add header that sends user to the issues page when clicked.

[why]
Header that redirects users to the main(issues) page always exists on top of the page.

[what]
App.js
index.js

[etc]

pje1740 committed:
[FIX] check null state for milestones in issue detail page.

[why]
The detail page did not load when the issue did not have milestone id.

[what]
We check whether the milestones are null or not, and if it is, we give empty array to MileSelected Data.

[etc]

JeesooHaa commented on Frontend/Views/Components/IssueDetail/IssueDetailWrapper.js:
```diff
@@ -28,15 +28,19 @@ const IssueDetailWrapper = ({ issueId }) => {
+      setMileSelectedData([]);
+    } else {
+      setMileSelectedData([result.data.issue.milestoneId]);
+    }
```
`else` 문 없이도 가능하지 않을까요...?

JeesooHaa commented on Frontend/Views/Components/NewIssue/NewIssueForm.js:
```diff
@@ -25,7 +26,6 @@ const NewIssueForm = ({ userSelectedData, labelSelectedData, mileSelectedData })
       title,
       comment,
       userId: getUserId(document.cookie),
```
이 부분은 이제 서버에서 처리해도 될 것 같아요.

JeesooHaa reviewed (approved):
말씀드린 부분 수정하고 머지하면 될 것 같습니다.

pje1740 committed:
[FIX] fix new issue post body, IssueDetailWrapper if/else

[why]
MileSelectedData is an empty array by default.
When creating new issue, the server can retrieve the user id.

[what]

[etc]

thesulks commented on Frontend/Views/Components/IssueDetail/IssueDetailWrapper.js:
```diff
@@ -18,7 +18,7 @@ const IssueDetailWrapper = ({ issueId }) => {
       });
     } catch (err) {
-      console.err(err);
+      console.err(err.message);
```
말씀한 부분 체크했습니다

thesulks commented on Frontend/Views/Components/IssueDetail/IssueDetailWrapper.js:
```diff
@@ -28,15 +28,17 @@ const IssueDetailWrapper = ({ issueId }) => {
       });
     } catch (err) {
-      console.err(err);
+      console.err(err.message);
```
체크2

thesulks reviewed (approved):
고생하셨습니다 👍
